custom/samveg/case_importer/validators.py

```diff
@@ -107,6 +110,43 @@ def run(cls, row_num, raw_row, fields_to_update):
         return fields_to_update, error_messages
 
 
+class UploadLimitValidator(BaseValidator):
+    @classmethod
+    def run(cls, row_num, raw_row, fields_to_update, import_context):
+        error_messages = []
+        owner_name = raw_row.get(OWNER_NAME)
+        call_value, call_number = cls._get_latest_call_value_and_number(raw_row)
+        if owner_name and call_number:
+            if cls._upload_limit_reached(import_context, owner_name, call_number):
+                error_messages.append(
+                    _('Limit reached for {owner_name} for call {call_number}').format(
+                        owner_name=owner_name,
+                        call_number=call_number
+                    )
+                )
+            else:
+                cls._update_counter(import_context, owner_name, call_number)
+        else:
+            error_messages.append(
+                _('Missing owner or call details')
+            )
+        return fields_to_update, error_messages
+
+    @classmethod
+    def _upload_limit_reached(cls, import_context, owner_name, call_number):
+        return cls._counter(import_context)[owner_name][call_number] >= ROW_LIMIT_PER_OWNER_PER_CALL_TYPE
+
+    @classmethod
+    def _update_counter(cls, import_context, owner_name, call_number):
+        cls._counter(import_context)[owner_name][call_number] += 1
```

Comment:
Nitpick: I might use the label (like `Call1` ) here instead of just the number. When troubleshooting & reading tests, it's odd to see hash keys that are plain numbers.

Reply:
https://github.com/dimagi/commcare-hq/pull/30928/commits/6462694b4166b7ee744d7fd6e4dca88ea1abbf59